### ISSUE TITLE
Public API proxy template doesn't set permission for service account

### ISSUE THREAD

**Issue by mkralik3** [syndesisio/syndesis#4702]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
I have a clean minishift instance with syndesis:
`syndesis minishift --full-reset --install --project syndesis --openshift-version 3.11.0`

I create template for public [endpoint](https://github.com/syndesisio/syndesis/tree/master/install#create-template-for-public-api-endpoint)
`oc create -f install/support/syndesis-public-oauth-proxy.yml`

After that, I create a new app
```
oc new-app --template=syndesis-public-oauthproxy \
    -p PUBLIC_API_ROUTE_HOSTNAME= public-syndesis.192.168.42.4.nip.io \
    -p OPENSHIFT_PROJECT=$(oc project -q) \
    -p OPENSHIFT_OAUTH_CLIENT_SECRET=$(oc sa get-token syndesis-oauth-client) \
    -p SAR_PROJECT=$(oc project -q)
```
The output looks good:
```
--> Deploying template "syndesis/syndesis-public-oauthproxy" to project syndesis

     syndesis-public-oauthproxy
     ---------
     Syndesis Public API is deployed to public-syndesis.192.168.42.4.nip.io.

     * With parameters:
        * PUBLIC_API_ROUTE_HOSTNAME=public-syndesis.192.168.42.4.nip.io
        * OpenShift project to deploy into=syndesis
        * OpenShift project to be used to authenticate the user against=syndesis
        * OPENSHIFT_OAUTH_CLIENT_SECRET=<TOKEN>
        * OAUTH_PROXY_TAG=v1.1.0
        * Syndesis Image Registry=docker.io
        * Image Stream Namespace=
        * OAuth Cookie Secret=ENHr4n1W4DEnYOliL4LdCGXuu0WBkunj # generated

--> Creating resources ...
    serviceaccount "syndesis-public-oauthproxy" created
    rolebinding.authorization.openshift.io "syndesis-public-oauthproxy:viewers" created
    clusterrolebinding.authorization.openshift.io "syndesis-syndesis-auth-delegator" created
    imagestream.image.openshift.io "syndesis-public-oauthproxy" created
    service "syndesis-public-oauthproxy" created
    route.route.openshift.io "syndesis-public-api" created
    deploymentconfig.apps.openshift.io "syndesis-public-oauthproxy" created
--> Success
    Access your application via route 'public-syndesis.192.168.42.4.nip.io' 
    Run 'oc status' to view your app.
```
But the syndesis-public-oauthproxy contains error in log:
```
unable to load OpenShift configuration: unable to retrieve authentication information for tokens: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:syndesis:syndesis-public-oauthproxy" cannot create tokenreviews.authentication.k8s.io at the cluster scope: no RBAC policy matched
```
It is strange because the [delegator](https://github.com/syndesisio/syndesis/blob/master/install/support/syndesis-public-oauth-proxy.yml#L75) is successfully created according to log.

After add  permission manually
```
oc adm policy add-cluster-role-to-user system:auth-delegator system:serviceaccount:syndesis:syndesis-public-oauthproxy
```
the public-oauthproxy pod works.

For the reproducing this, it is necessary to have a fresh minishift instance. (--full-reset)

**Comment by dhirajsb** [comment]
@mkralik3 can you verify that the user you use to create Syndesis has cluster admin permissions from the beginning?

**Comment by mkralik3** [comment]
Yes, after installing, I login as `oc login -u system:admin`

**Comment by dhirajsb** [comment]
Thanks. I will test with a fresh install of minishift on my machine again.

**Comment by dhirajsb** [comment]
I ran into some trouble installing minishift from scratch today, had to update some dependencies since I upgraded to Mojave. I am hoping to test again tomorrow and fix it if I can recreate this issue.

**Comment by dhirajsb** [comment]
I finally managed to more or less run your exact commands and it looks like the issue is in the clusterrolebinding it creates:
```
...
Subjects:
  Kind  Name                                                           Namespace
  ----  ----                                                           ---------
  User  system:serviceaccount:${NAMESPACE}:syndesis-public-oauthproxy
```
It's not injecting the namespace correctly.

**Comment by dhirajsb** [comment]
@mkralik3 please close this once you have verified the fix from PR #4798
